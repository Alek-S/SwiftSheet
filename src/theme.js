/**
 * default themes for use with styled-components
 * passed in with theme provider
 */
const theme = {
	boxShadow: `0px 5px 40px -5px rgba(0,0,0,0.3)`,
	boxShadowDark: `0px 5px 40px -2px rgba(0,0,0,0.5)`,
	color: {
		background: '#fafafa',
		backgroundDark: '#EAEAEA',
		backgroundDarkest: '#BCBCBC',
		border: '#d9d9d9',
		text: '#404040',
		lightText: '#848484',
		red: '#ED5050',
		lightRed: '#FF8D8D',
		green: '#D8E13F',
		blue: '#46C8F5',
		lightBlue: '#97E4FF',
		darkBlue: '#2E8AA8',
	},
	gradient: {
		light: `linear-gradient(to bottom, #ffffff 0%, #f6f6f6 47%, #ededed 100%)`,
		red: 'linear-gradient(to bottom, #f07878 0%, #ED5050 100%)',
		lightRed: 'linear-gradient(to bottom, #f39191 0%, #ef6262 100%)',
		blue: 'linear-gradient(to bottom, #D8E13F 0%, #BFC644 100%)',
		greenBlue: ' linear-gradient(160deg, #57DACB 0%, #55A2ED 100%)',
	},
	font: {
		header: `'Poiret One', cursive`,
		main: `'Quicksand', sans-serif`,
	},
};

export default theme;
