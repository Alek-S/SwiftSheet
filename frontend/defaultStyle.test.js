import React from 'react';
import 'jest-styled-components';
import renderer from 'react-test-renderer';

import theme from './theme';
import DefaultStyle, { ApiCard, Policy } from './defaultStyle';

describe('defaultStyle', () => {
	const tree = renderer.create(<DefaultStyle theme={theme} />);

	test('matches the snapshot', () => {
		expect(tree).toMatchSnapshot();
	});

	// test('has correct background', () => {
	// 	expect(tree).toHaveStyleRule('background-color', 'white');
	// });

	// test('text has correct theme color', () => {
	// 	expect(tree).toHaveStyleRule('color', theme.color.text);
	// });

	// test('text has correct theme font-family', () => {
	// 	const font = theme.font.main.replace(/\s/g, '');
	// 	expect(tree).toHaveStyleRule('font-family', font);
	// });

	// test('text has correct font-size', () => {
	// 	expect(tree).toHaveStyleRule('font-size', '0.9em');
	// });
});

describe('ApiCard', () => {
	const tree = renderer.create(<ApiCard theme={theme} />).toJSON();

	test('matches the snapshot', () => {
		expect(tree).toMatchSnapshot();
	});
});

describe('Policy', () => {
	const tree = renderer.create(<Policy theme={theme} />).toJSON();

	test('matches the snapshot', () => {
		expect(tree).toMatchSnapshot();
	});
});
