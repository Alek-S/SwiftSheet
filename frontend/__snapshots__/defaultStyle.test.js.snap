// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApiCard matches the snapshot 1`] = `
<div
  className="defaultStyle-g21vfe-0 defaultStyle__Card-g21vfe-1 defaultStyle__ApiCard-g21vfe-4 izQYFc"
/>
`;

exports[`Policy matches the snapshot 1`] = `
<section
  className="defaultStyle__Policy-g21vfe-8 dptqDY"
/>
`;

exports[`defaultStyle matches the snapshot 1`] = `
<div
  className="defaultStyle-g21vfe-0 dMrRuR"
/>
`;
